Foundation/Scanner.swift

```diff
@@ -72,6 +70,89 @@ open class Scanner: NSObject, NSCopying {
         _skipSet = Scanner.defaultSkipSet
         _scanLocation = 0
     }
+    
+    // On overflow, the below methods will return success and clamp
+    @discardableResult
+    open func scanInt32(_ result: UnsafeMutablePointer<Int32>) -> Bool {
```

Comment:
Am I misreading the Scanner documentation or should all of these  `scanXXX(_ result: UnsafeMutablePointer<XXX>) -> Bool` methods actually take an optional pointer?
https://developer.apple.com/documentation/foundation/scanner/1410914-scanint32

Looks like they may have always been wrong?

Reply:
Yeah, good catch. I'll follow up to correct the signatures.